### PR TITLE
Aden-13832 ATF vs BTF slots auctions split  

### PR DESCRIPTION
**Pull request by lnawrotek**

### CODE REVIEW

File: src/ad-bidders/bidder-provider.ts [hunk 1]
```diff
@@ -54,6 +54,14 @@ export abstract class BidderProvider {
 		utils.logger(this.logGroup, 'respond');
 	}
 
+	protected onNoSlotsError(): void {
```
Comment (preludek) [inline src/ad-bidders/bidder-provider.ts:57]: In both places this method is used there's already a utils logger that there are no slots to fetch bids. We can either remove it from here or where this function is called

File: src/platforms/shared/setup/bid-auction-split-experiment-setup.ts
Comment (preludek) [inline src/platforms/shared/setup/bid-auction-split-experiment-setup.ts:1]: We used to have a separate path for experiments. `../experiments` i think 🤔 I suggest moving it either way so it's easier to find and remove later on

**Review by preludek — DISMISSED** [review]

**Review by preludek — DISMISSED** [review]

File: src/ad-bidders/bidder-provider.ts [hunk 2]
```diff
@@ -54,6 +54,12 @@ export abstract class BidderProvider {
...
```
Comment (adi-wesolowski-fandom) [inline src/ad-bidders/bidder-provider.ts:57]: nitpick: "on" suggest some kind of listener. Maybe "markNoSlots"?

File: src/ad-bidders/prebid/prebid-data-tracking.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			(response: { adId: string; bid: PrebidBidResponse; doc: Document | null }) =>
+				intentIQ.reportPrebidWin(response.bid),
+		);
+
+		utils.logger(logGroup, 'Bids tracking configured');
+	}
+
+	private enableATSAnalytics(): void {
+		if (
+			context.get('bidders.liveRampATSAnalytics.enabled') &&
```
Comment (adi-wesolowski-fandom) [inline src/ad-bidders/prebid/prebid-data-tracking.ts:85]: nitpick: quick return would be more clear to read. And those conditions can be merged into one function "isLiveRampEnabled()"

File: src/platforms/shared/experiments/bid-auction-split-experiment-setup.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	call(): void {
+		this.setupExperimentVariants();
+		this.activeExperimentVariant = getExperiment(this.experimentVariants);
+
+		if (this.isExperimentEnabled()) {
+			addExperimentGroupToTargeting(this.activeExperimentVariant.name);
+
+			if (this.isControlVariant()) {
+				utils.logger(logGroup, 'Experiment but control group', this.activeExperimentVariant.name);
```
Comment (adi-wesolowski-fandom) [inline src/platforms/shared/experiments/bid-auction-split-experiment-setup.ts:35]: nitpick: here also, deep nesting. With quick return of the first condition, there will be one indent less

File: src/platforms/shared/experiments/bid-auction-split-experiment-setup.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+			} else {
+				utils.logger(logGroup, 'Experiment - active group', this.activeExperimentVariant.name);
+				context.set('custom.bidAuctionSplitEnabled', true);
+			}
+		}
+	}
+
+	private setupExperimentVariants() {
+		if (this.isMobile) {
+			this.experimentVariants = [
```
Comment (adi-wesolowski-fandom) [inline src/platforms/shared/experiments/bid-auction-split-experiment-setup.ts:45]: nitpick: we got it as class property, so maybe extract these two condition result blocks, into separate function? This also would reduce nesting :)

**Review by adi-wesolowski-fandom — APPROVED** [review]

File: src/ad-bidders/bidder-provider.ts (hunk 2 above)
Comment (lnawrotek) [reply src/ad-bidders/bidder-provider.ts:57]: It's true, but I named this function so that the name would be similar to the existing "onBidResponse"

**Review by adi-wesolowski-fandom — APPROVED** [review]

**Review by adi-wesolowski-fandom — APPROVED** [review]

**Review by adi-wesolowski-fandom — DISMISSED** [review]

**Review by preludek — APPROVED** [review]

**Review by preludek — DISMISSED** [review]

**Review by Jorewin — APPROVED** [review]

**Pull request merged**